### PR TITLE
Add Ruuvi and BTHome support (Shelly BLE devices)

### PR DESCRIPTION
**Pull request by gusgonnet**
This PR adds Ruuvi tags and BTHome support.
It also fixes the build on Photon2/P2.

https://ruuvi.com/
https://bthome.io/

BTHome support enables Shelly BLE devices like:
[BLE Button](https://www.shelly.com/en/products/shop/shelly-blu-button1)
[BLE window/door sensor
](https://www.shelly.com/en/products/shop/shelly-blu-door-window)

Example projects to use them:

https://www.hackster.io/gusgonnet/ble-door-window-sensor-12f054
https://www.hackster.io/gusgonnet/ble-door-window-alerts-432429
https://www.hackster.io/gusgonnet/ble-button-for-your-photon-2-fdc8a8

### CODE REVIEW

File: src/BTHome.cpp [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        Log.error("BTHome: advertisement parsing failed");
+    }
+}
+
+bool BTHome::isBeacon(const BleScanResult *scanResult)
+{
+    uint8_t buf[BLE_MAX_ADV_DATA_LEN];
+    uint8_t count = ADVERTISING_DATA(scanResult).get(BleAdvertisingDataType::SERVICE_DATA, buf, BLE_MAX_ADV_DATA_LEN);
+
+    if (count > 3 && buf[0] == 0xD2 && buf[1] == 0xFC) // BTHome UUID is 0xFCD2
```
Comment (eberseth) [inline src/BTHome.cpp]: Please create an inline function to check for the UUID byte sequence belongs to this family.  I noticed the magic numbers here are also used elsewhere in this file.

File: src/BTHome.cpp [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+bool BTHome::parseBTHomeAdvertisement(const uint8_t *buf, size_t len)
+{
+
+    if (len < 9)
+    {
+        return false;
+    }
+
+    // first two bytes are the UUID
+    if (buf[0] != 0xD2 || buf[1] != 0xFC)
```
Comment (eberseth) [inline src/BTHome.cpp]: See comment above

File: src/BTHome.cpp [hunk 3]
```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2020 Particle Industries, Inc.
```
Comment (eberseth) [inline src/BTHome.cpp]: nit: 2024

File: src/BTHome.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020 Particle Industries, Inc.
```
Comment (eberseth) [inline src/BTHome.h]: nit: 2024

File: src/BTHome.cpp (hunk 1 above)
Comment (gusgonnet) [reply src/BTHome.cpp:41]: Hey Eric, can you elaborate a bit in the shape and form of that inline function, please? I'm not familiar. 
Is there an example elsewhere in the code of that?
Thanks

**Comment by gusgonnet** [comment]
@eberseth I have addressed your comments in my last commit. Let me know how it looks now.
Thanks

File: src/BTHome.cpp (hunk 1 above)
Comment (eberseth) [reply src/BTHome.cpp]: For example:
```cpp
static inline bool BTHome::isBTHome(uint8_t lsb, uint8_t msb) const { return (0xD2 == lsb) && (0xFC == msb); } // BTHome UUID is 0xFCD2

if (count > 3 && isBTHome(buf[0], buf[1])) // BTHome UUID
```

Comment (eberseth) [reply src/BTHome.cpp]: That would be reused on line 101

Comment (gusgonnet) [reply src/BTHome.cpp]: got it, I addressed this in my next commit. Thanks

File: src/ruuvi.cpp [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        Log.error("Ruuvi: advertisement parsing failed");
+    }
+}
+
+bool Ruuvi::isBeacon(const BleScanResult *scanResult)
+{
+    uint8_t buf[MAX_MANUFACTURER_DATA_LEN];
+    uint8_t count = ADVERTISING_DATA(scanResult).get(BleAdvertisingDataType::MANUFACTURER_SPECIFIC_DATA, buf, MAX_MANUFACTURER_DATA_LEN);
+
+    if (count > 3 && buf[0] == 0x99 && buf[1] == 0x04) // Ruuvi UUID is 0x9904
```
Comment (eberseth) [inline src/ruuvi.cpp]: Similar to the BTHome case:

Also, double check for endianess.

```cpp
static inline bool Ruuvi::isRuuvi(uint8_t lsb, uint8_t msb) const { return (0x99 == lsb) && (0x04 == msb); } // Ruuvi UUID is 0x9904

if (count > 3 && isRuuvi(buf[0], buf[1])) // Ruuvi UUID
```

**Comment by gusgonnet** [comment]
@eberseth I have addressed your second round of comments. Thanks!

**Review by eberseth — APPROVED** [review]

**Comment by gusgonnet** [comment]
Hi @eberseth @Dan-Kouba ,
I may not have permissions to merge this PR, can you please do it?
Thanks

**Pull request merged**